### PR TITLE
fix(storage):  batch qurry with_table_option

### PR DESCRIPTION
**Pull request by Li0k**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

as title, some executor use `StorageTable::new_partial` to initialize `StorageTable` without `TableOption`, need `retaintion_seoncd` properties for user_iterator to guarantee data visibility

- `TableDesc` add `retaintion_second` field
- `new_partial` interface adapt retaintion_second for read_option
- rename `ttl` to `retaintion_second` for user understandability

## Checklist

- ~[ ] I have written necessary rustdoc comments~
- ~[ ] I have added necessary unit tests and integration tests~
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)
- https://github.com/singularity-data/risingwave/issues/3298

### CODE REVIEW

File: src/frontend/src/catalog/table_catalog.rs [hunk 1]
```diff
@@ -115,6 +116,11 @@ impl TableCatalog {
 
     /// Get a [`TableDesc`] of the table.
     pub fn table_desc(&self) -> TableDesc {
+        use risingwave_common::catalog::TableOption;
+        // use risingwave_common::config::constant::hummock::PROPERTIES_RETAINTION_SECOND_KEY;
```
Comment (wcy-fdu) [inline src/frontend/src/catalog/table_catalog.rs:120]: remove this?

**Review by wcy-fdu — APPROVED** [review]
I think most of changes are renamed, LGTM!

File: proto/hummock.proto [hunk 2]
```diff
@@ -152,7 +152,7 @@ message KeyRange {
 }
 
 message TableOption {
-  uint32 ttl = 1;
+  uint32 retaintion_second = 1;
```
Comment (hzxa21) [inline proto/hummock.proto:155]: should be `retention_seconds`

Comment (hzxa21) [reply proto/hummock.proto]: The naming is still wrong. "retention" instead of "retaintion"

**Review by hzxa21 — APPROVED** [review]
Please fix the naming.

Rest LGTM.

**Comment by Li0k** [comment]
> Please fix the naming.
> 
> Rest LGTM.

I make some mistake at last pr, fix it now , thanks for review

**Pull request merged**